packages/core/src/core/pantheon-api.ts

```diff
@@ -202,8 +202,12 @@ export const PantheonAPI = (givenOptions?: PantheonAPIOptions) => {
           return res.redirect(302, options.notFoundPath);
         }
 
-        // Fetch the site
-        const site = await PCCConvenienceFunctions.getSite();
+        // Define site
+        let site: Site | undefined;
+        // Fetch the site if we dont use a pccGrant
+        if (!pccGrant) {
+          site = await PCCConvenienceFunctions.getSite();
```

Comment:
It'd be good for us to parallelise these. We should also use the client provided in `options.getPantheonClient` for fetching the site, PCCConvenienceFunctions doesn't use this

```suggestion
            const client = options.getPantheonClient({
          pccGrant: pccGrant ? pccGrant.toString() : undefined,
        });

        const [article, site] = await Promise.all([
          parsedArticleId == null
            ? null
            : await getArticleBySlugOrId(
                client,
                parsedArticleId,
                // We will let downstream validate the publishingLevel param.
                {
                  publishingLevel: publishingLevel
                    ?.toString()
                    .toUpperCase() as AllowablePublishingLevels,
                },
              ),
          !client.apiKey?.startsWith("pcc_grant")
            ? // Fetching the site is not available when the client is using a PCC grant.
              getSite(client, client.siteId)
            : null,
        ]);
```